Assets/_BForBoss/_Weapons/Scripts/Bullets/BulletBehaviour.ApplyDamage.cs

```diff
@@ -1,13 +1,18 @@
-using Perigon.Entities;
 using UnityEngine;
 
 namespace Perigon.Weapons
 {
+    public interface ILifeCycleBehaviour
```

Comment:
Because i'm making the Entity depend on the weapon, Unity complains about a cycle dependency 
Weapon Depends on Entity
Entity Depends on Weapon. 

So I can either choose one. So because Weapon could be used in Enemies And Players, I chose Weapon to be the universal one. 

So right now, Entities depend on weapons. So because Weapon is universal, it has this `ILifeCycleBehaviour`. Whoever, or whatever needs this weapon, will need to implement this `ILifeCycleBehaviour`, to use weapons. 

TLDR - Bullet need to cause damage to thing. so as long as the thing needs to weapons, it also needs a life cycle.